### PR TITLE
Makes Medibeam researchable

### PR DESCRIPTION
**Pull request by Robertha89**
Allows science to build Medibeam guns

Ubercharge Implants when?

:cl: PPI
rscadd: Science can now research the Medical Beamgun.
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
>just reusing the same one

make it biotech 8 so its not every goddamn round

File: code/modules/projectiles/guns/medbeam.dm [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /obj/item/weapon/gun/medbeam
 	name = "Medical Beamgun"
 	desc = "Delivers medical nanites in a focused beam."
+	origin_tech = "combat=6;materials=7;powerstorage=8;programming=7;biotech6"//The ultimate within medical design
```
Comment (RemieRichards) [inline code/modules/projectiles/guns/medbeam.dm]: you forgot the equals for biotech.

**Comment by LatD** [comment]
It's very rare to see both bio 6 and combat 6 in research. Also programming 7 can only be get with experimentor.

**Comment by Bawhoppen** [comment]
you can get data 7 from phazon circuits

this should be rare to get

**Comment by LatD** [comment]
Phazon circuits have origin tech 5 so they only get it to 6. Nothing can get it to 7 as far as I know.

File: code/modules/projectiles/guns/medbeam.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/medbeam.dm]: Personally, I don't think these are good numbers; the super ultra high-tier items generally have lower origin tech than what is required to research them.

These will end up being valuable to scientists not just for "I get to heal everything and invalidate medbay!" but also because they'll be able to hit materials 8 and programming 8---meaning even cheaper R&D items and even faster mech production.

**Comment by Bawhoppen** [comment]
well decrease the data tech to 6 then and increase biotech to 8

**Comment by Fox-McCloud** [comment]
Didn't Goof or someone else plan to make a backpack version of this, complete with slowdown and such, an actual R&D item? This comes across as awfully powerful to just let any crewmember have.

File: code/modules/research/designs/medical_designs.dm [hunk 2]
```diff
@@ -118,6 +118,17 @@
...
 	category = list("Medical Designs")
 
+/datum/design/medibeam
+	name = "Medical Beamgun"
+	desc = "Delivers medical nanites in a focused beam."
+	id = "medi_beam"
+	req_tech = list("combat" = 6, "materials" = 7, "powerstorage" = 8, "programming" = 7, "biotech" = 6)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 4000, MAT_URANIUM = 8000, MAT_DIAMOND = 2000)
+	reliability = 100
```
Comment (Fox-McCloud) [inline code/modules/research/designs/medical_designs.dm]: base datum reliability is already 100.

**Comment by Bawhoppen** [comment]
when I suggested it to anturk originally I wanted a backpack to go with it
I guess we should make it have one now

File: code/modules/research/designs/medical_designs.dm [hunk 3]
```diff
@@ -118,6 +118,17 @@
 	build_path = /obj/item/device/extinguisher_refill
 	category = list("Medical Designs")
 
+/datum/design/medibeam
+	name = "Medical Beamgun"
+	desc = "Delivers medical nanites in a focused beam."
+	id = "medi_beam"
+	req_tech = list("combat" = 6, "materials" = 7, "powerstorage" = 8, "programming" = 7, "biotech" = 6)
```
Comment (GunHog) [inline code/modules/research/designs/medical_designs.dm]: The highest obtainable programming (Data Theory) level is 6, using Phazon boards. The exception to this is of course the Experimentor, which can theoretically unlock anything. (Although trying to getting more than one tech level up on the same object has astronomically low odds)

I recommend dropping programming down one level, as Combat 6 (Combat Shotguns) and Bio 6 (Sentience Potion) are tough gates to crack internally.

**Comment by GunHog** [comment]
I loved this gun during my first challenge Ops round. I got shot to crit through laser fire, and a fellow Op got me on my feet again so I could get a few more kills. We fluked; The crew escaped on the shuttle, but I did survive thanks to that.

**Comment by Bawhoppen** [comment]
I again recommend we should have it biotech 8, so its slightly difficult to get but by no means impossible

**Comment by Robertha89** [comment]
the only way to get it to 6 or 7 is to kill get a wumborian fugu gland, or two for 7 as it is, excluding that, there is the firearms authentication implant. 
It is currently hard enough to get the rest of science to that level, making it 8 would make it completely rng based, as you need either a changeling, a xenomorph, or a slaughter demon to get it.

**Comment by Bawhoppen** [comment]
an infinite extremely quick ranges heal? needs to be hard to get

**Comment by TheNightingale** [comment]
It still needs to be achievable. At 6, your best hope is either the rare fugu gland, or RNG Experimentor for an hour. Bio 6 is high enough.

**Comment by Bawhoppen** [comment]
I think it should be required to get an alien brain or something else rarer

**Comment by KorPhaeron** [comment]
Does everything in the game need to end up in RD?

**Comment by Bawhoppen** [comment]
@KorPhaeron that was my thoughts precisely

But I do think having a medigun of some kind would be cool for science

**Comment by GunHog** [comment]
Requirements are great as they are! CMO would orgasm to have an item like that

**Comment by Zilenan91** [comment]
Put it in mining loot rooms and call it good. If not that this need to be bio 8 or something because this is too powerful for normal rounds.

**Comment by LatD** [comment]
Getting even the current requirement levels needs a round over one hour long and a science team full of good scientists. It's very rare to get both bio 6 and combat 6. I mean if you're really trying to get it it's possible faster but usually that high levels are only got when all things just happen to go perfectly (someone good is in xenobio, cargo is providing combat shotguns, someone good is doing R&D, mining is actually working etc). I'd say it's good as it is now, a rare really high tech item that will be seen on rounds where all departments are being good.

**Comment by Lumipharon** [comment]
Are you guys having a giggle?

Getting those levels are pathetically easy.
The experimentor is so fucking over powered it's not even funny. You just print out a dozen of the items (durand weapon board etc) and just spam it until you get blood drain, have a medibot outside to heal you, and bam, 2-5 minutes you get combat 6.
And once you have combat 6, you can get illegal tech up to 4, which gives you in the process bio 7 (thermal implant > xray implant).

Even bio 8 isn't terrible hard to get.

TL;DR: The experimentor makes all of R&D laughably easy (except for shit like combat 8 for decloners).

**Comment by Iamgoofball** [comment]
I'll give this a try with a backpack locked variant.

**Comment by Bawhoppen** [comment]
make the synde one need a backpack too for consistency, but make it better

**Comment by Zilenan91** [comment]
That'd be great if the syndi one was flat-out better. Oh and can we please make this bio 8? I don't want insta-heal beams spammed every single round

**Comment by Bawhoppen** [comment]
syndies also have plenty of room even with out a backpack
just integrate their jetpack into it

**Comment by GunHog** [comment]
>syndies also have plenty of room even with out a backpack
just integrate their jetpack into it

Did you know that standard Syndicate Hardsuits actually DO have built in jetpacks? Syndies just have no way of using them, as they lack access to large air tanks, and using their box air tank as fuel is just plain daft, as they need it for internals. Also, the suit does not accept the jetpack harness into the suit storage.

**Comment by Bawhoppen** [comment]
oh ye just give em some tanks ez pz

**Comment by GunHog** [comment]
If you do so, Baw, make sure to set jetpacks on the other buyable Syndie hardsuits.

**Comment by Iamgoofball** [comment]
Syndie one will stay without needing a backpack because >making nuke ops inventory management even worse

**Comment by Bawhoppen** [comment]
If you have a medigun you're healing people, not the front lines
You still have like 8 slots for guns

**Comment by Zilenan91** [comment]
You can't hold primaries anywhere but your bag anyways but yeah that's probably fine unless we want to make it so they can put guns on their belt.

**Comment by Man-Shroom-** [comment]
Booooo not another research buff, it should be orderable in cargo or something and not an RnD item

**Comment by Dorsisdwarf** [comment]
0/10 Let adminspawns be adminspawns

**Comment by duncathan** [comment]
give the cool things to another department
